### DISCUSSION TITLE
How to write router function in nuxtJS

### DISCUSSION THREAD

**Discussion by Leopard-CC** [community/community#36217] [Other Feature Feedback, Questions, & Ideas] [answered]
I am using Nuxt.js, but I got a problem in router's functions.

In the pure Vue, i can write in main.js like this:

```
val route = new Router({
   routes:{
      [...]
   }
})

route.beforeEach(to,from,next){
    //do something to validate
}
```
And how to do the same in nuxt.js ? I can not find any file like main.js.

Also, all I know is to deal with the pages folder to achieve router, I can not set the redirect path

please help, thx :)

**Comment by Birdza87** [comment]
Hi

**Comment by softdream815** [comment]
create a plugins/route.js file:

export default ({ app }) => {
   // Every time the route changes (fired on initialization too)
   app.router.afterEach((to, from) => {
     //do something to validate
   })
}
and update your nuxt.config.js file:

plugins: ['~/plugins/route']
More details about Nuxt plugins: https://nuxtjs.org/guide/plugins

**Reply by Leopard-CC** [reply]
It worked for me, You saved the day!
Thank you mate

**Comment by raihanhafiz2015** [comment, marked as answer]
Hi,
Please check this link
https://nuxtjs.org/guide/pluginsprimary
